### PR TITLE
Use new metric infrastructure for metric publishing

### PR DESCRIPTION
**Pull request by justinvdm**
The plan is to use the metric infrastructure added in #753 to replace the current way of metric publishing in Go.

### CODE REVIEW

**Comment by justinvdm** [comment]
Ready for review.

File: go/vumitools/app_worker.py [hunk 1]
```diff
@@ -280,25 +268,21 @@ def trigger_event(self, message, event_type, content):
...
+    def publish_conversation_metric(self, conv, name, value, agg=None):
+        # NOTE: The plan is have conversation metrics specified as part of the
+        # conversation definition, so not letting them be dynamically created
+        # is intentional. This workaround will go away very soon.
+        class DynamicConversationMetric(ConversationMetric):
+            METRIC_NAME = name
+            AGGREGATORS = [agg] if agg else ConversationMetric.AGGREGATORS
+
+        metric = DynamicConversationMetric(conv)
+        metric.oneshot(self.metrics, value)
```
Comment (justinvdm) [inline go/vumitools/app_worker.py:280]: This uglyness is a temporary workaround and will go away in #750 (what I will be working on next).

File: go/base/middleware.py [hunk 2]
```diff
@@ -34,21 +31,23 @@ class ResponseTimeMiddleware(object):
...
-            stop_time = time.time()
-            func = resolve(request.path)[0]
-            metric_name = '%s.%s.%s' % (func.__module__, func.__name__,
-                                        request.method)
-            response_time = stop_time - request.start_time
-            self.publish_metric(metric_name.lower(), response_time)
-            response['X-Response-Time'] = response_time
+            metric = self.metric_from_request(request)
+            metric.oneshot(response_time)
         except AttributeError, e:
```
Comment (hodgestar) [inline go/base/middleware.py:51]: I wonder what this `AttributeError` was supposed to catch? I'm _guessing_ it was `request.start_time`.

**Comment by hodgestar** [comment]
Other than the question about what should be in the `AttributeError` exception, :+1:.

File: go/base/middleware.py (hunk 2 above)
Comment (justinvdm) [inline go/base/middleware.py:51]: Hmm, probably. That makes sense, so I think it is still relevant. I'll add a comment.

Comment (justinvdm) [inline go/base/middleware.py:51]: Added a comment, not sure if anything else should be changed.

Comment (hodgestar) [inline go/base/middleware.py:51]: `request.start_time` is currently accessed outside the `try: ... except: AttributeError` block. :)

Comment (justinvdm) [inline go/base/middleware.py:51]: Doh :) Thanks, will fix

**Comment by justinvdm** [comment]
Ready for another review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**